store/driver/txn/batch_getter.go

```diff
@@ -16,11 +16,60 @@ package txn
 
 import (
 	"context"
+	"unsafe"
 
-	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb/kv"
+	tikverr "github.com/tikv/client-go/v2/error"
+	"github.com/tikv/client-go/v2/txnkv/transaction"
 )
 
+// tikvBatchGetter is the BatchGetter struct for tikv
+// In order to directly call NewBufferBatchGetter in client-go
+// We need to implement the interface (transaction.BatchGetter) in client-go for tikvBatchGetter
+type tikvBatchGetter struct {
+	tidbBatchGetter BatchGetter
+}
+
+func (b tikvBatchGetter) BatchGet(ctx context.Context, keys [][]byte) (map[string][]byte, error) {
+	// toTiDBKeys
+	kvKeys := *(*[]kv.Key)(unsafe.Pointer(&keys))
+	vals, err := b.tidbBatchGetter.BatchGet(ctx, kvKeys)
+	return vals, err
+}
+
+// tikvBatchBufferGetter is the BatchBufferGetter struct for tikv
+// In order to directly call NewBufferBatchGetter in client-go
+// We need to implement the interface (transaction.BatchBufferGetter) in client-go for tikvBatchBufferGetter
+type tikvBatchBufferGetter struct {
+	tidbMiddleCache Getter
+	tidbBuffer      BatchBufferGetter
+}
+
+func (b tikvBatchBufferGetter) Get(k []byte) ([]byte, error) {
```

Comment:
Where would this method be used? Seems we're refactoring the `batchGet` in this PR?

Reply:
1、`BatchGet` method in client-go will call this method. Like this: https://github.com/tikv/client-go/blob/b11da6307c6fc761e0d8841a72bccd578be1ef82/txnkv/transaction/batch_getter.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L75
2、The purpose of this PR is to remove the repeated implementation of `BatchGet` method in `TiDB`, and directly use `BatchGet` method in `client-go`.